### ISSUE TITLE
Possibility to set custom kvm parameters on an instance

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#317]
Originally reported of Google Code with ID 261.

```
It would be great to be able to add custom options to KVM on specific instances (or the entire cluster)

Examples I can think of:
 * adding -cpu host
 * Setting SMP sockets/cores/threads layout
 * Using virtfs via the -virtfs option
 * Anything else kvm can do but ganeti doesn't understand yet.

Proposed syntax might be:
gnt-instance modify -H kvm_flags="...."
gnt-cluster modify -H kvm:kvm_flags="..."


```
Originally added on 2012-08-11 11:24:11 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-10-30 11:13:43 +0000 UTC.

Added Labels: Type-Enhancement Hypervisor-KVM

**Comment by Ganeti-Issues-Migrator** [comment]
```
Windows 7 Professional supports only 2 sockets, but many cores/thrads per socket.

so it wold be cool to have the possibility to do something like:

gnt-instance modify -B vcpus="2,cores=2,threads=2" NAME

Thanks
```
Originally added on 2012-11-23 12:49:32 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-01-11 17:37:25 +0000 UTC.

Changed State: Accepted
Added to Milestone: Release2.7

**Comment by Ganeti-Issues-Migrator** [comment]
```
This has been fixed in:

4888a60 kvm: allow setting smp cores, threads, sockets

```
Originally added on 2013-02-26 15:55:55 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking 2.7 bugs fixed, as 2.7.0~rc1 was released, and should be ready to be tested in production environments.
```
Originally added on 2013-05-09 05:42:48 +0000 UTC.

Changed State: Released
